### PR TITLE
Upgrade WixToolset to v4.0.1

### PR DESCRIPTION
**Pull request by BMurri**
#### Details

Upgrade WixToolset to v4.0.1

##### Motivation

Addresses issue #1641 

##### Context

In WiX v4, it's easier than ever to build MSIs for each platform (x86, x64, arm64) using the exact same sources (with no conditional code).

The default build output directory for the MSI changes from `bin\$(Configuration)\...` to `bin\$(Platform)\$(Configuration)\...` where Platform = `x86`

This MSI contains suboptimal authoring, especially concerning the Windows Installer Best Practices relating to Component/File relationships. Generally speaking, each Component should have only one File. You can create ComponentGroups to group components as needed, but ComponentGroups don't survive in the MSI (they are an authoring-only concept).

It is NOT considered a best practice to mess with any per-user data (configuration or otherwise) in per-machine installation packages. The primary reason is that it is extremely difficult to touch all the user profiles that may have been affected, and you WILL leave configuration behind especially when a different user removes the package from the user that installed it. It would be a very good idea to review your custom actions that touch profile data.

`AccessibilityInsights.CustomActions.Package` can probably be merged into `AccessibilityInsights.CustomActions` eliminating the need for `OverwriteWithSignedAssemblies`. Custom action projects are no longer non-standard projects, they are now modern SDK-style projects, like any other csproj that can host either net-core or net-framework code.

#### Pull request checklist

- [x] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/main/docs/Scenarios.md) completed?
- [n/a] Does this address an existing issue? If yes, Issue# - 
- [n/a] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

**Comment by DaveTryon** [comment]
I'll need to create a test a signed build before approving this PR

**Comment by DaveTryon** [comment]
@bmurri, when I open the solution in Visual Studio 2022, the MSI project refuses to load. If I explicitly load it, it comes back with an error "Value does not fall within the expected range". Any thoughts on this? I have the [WiX 3.11.2 Visual Studio extension](https://marketplace.visualstudio.com/items?itemName=WixToolset.WiXToolset) installed--loading the solution without the extension produces an error about the project being incompatible with Visual Studio, so it seems to be required.

**Comment by BMurri** [comment]
I forgot to update the docs. You need the new HeatWave visual studio extension.
https://wixtoolset.org/docs/intro/
https://www.firegiant.com/wix/heatwave/
https://marketplace.visualstudio.com/items?itemName=FireGiant.FireGiantHeatWaveDev17 (for vs2022, there's also one for vs2019)

File: docs/SetUpDevEnv.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
    - Add ".NET Framework 4.8 development tools"
 
 ### Install Wix tools
-To build the MSI project, you need to install the Wix Toolset extension for Visual Studio. The extension is available in the Visual Studio marketplace and can be found [here](http://wixtoolset.org/releases/).
+To build the MSI project, you need to install the HeatWave extension for Visual Studio from FireGiant. The extension is available in the Visual Studio marketplace and can be found [here](http://wixtoolset.org/docs/intro/#vs).
```
Comment (codeofdusk) [inline docs/SetUpDevEnv.md:11]:
```suggestion
To build the MSI project, you need to install the HeatWave extension for Visual Studio from FireGiant. The extension is [available in the Visual Studio marketplace](http://wixtoolset.org/docs/intro/#vs).
```

**Review by codeofdusk — APPROVED** [review]
Please correct the path in `readme` line 43 to include the architecture if relevant.

**Comment by DaveTryon** [comment]
Thanks, @BMurri! Merging this now

**Pull request merged**